CRM/MembershipExtras/Job/OfflineAutoRenewal/PaymentPlan.php

```diff
@@ -741,4 +745,34 @@ protected function recordPaymentPlanFirstContribution() {
     }
   }
 
+  /**
+   * Checks if given line item already exists, by checking if there is already a
+   * line item with same entity_table, entity_id, contribution_id,
+   * price_field_value_id, and price_field_id.
+   *
+   * @param array $lineItem
+   *   Data for the line item to be used to check if it already exists.
+   *
+   * @return bool
+   *   TRUE if it finds a line item with the same combination of fields, FALSE
+   *   otherwise.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function isDuplicateLineItem($lineItem) {
+    $result = civicrm_api3('LineItem', 'get', [
+      'entity_table' => $lineItem['entity_table'],
```

Comment:
what of any one of these values is empty ,  these keys need to be checked before used.